### PR TITLE
[state][doc] Improve ray log doc

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: Ricky Xu <xuchen727@hotmail.com>

## Why are these changes needed?

- Improve clarity on `ray logs` and `ray logs cluster` alias. 
- Add example of `ray logs actor` 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-observability/state/state-api.rst [hunk 1]
```diff
@@ -449,7 +449,10 @@ E.g., Get a particular log file from a node
 
     .. code-block:: bash
 
-        # You could get the node id / node ip from `ray list nodes` 
+        # You could get the node id / node ip from `ray list nodes`
+        ray logs cluster gcs_server.out --node-id <NODE_ID>
+        # Or simply `ray logs ...` since `ray logs cluster` is alias to `ray logs`
```
Comment (rkooo567) [inline doc/source/ray-observability/state/state-api.rst:454]:
```suggestion
      #`ray logs cluster` is alias to `ray logs`
```

Comment (rkooo567) [reply doc/source/ray-observability/state/state-api.rst:454]: Can you also update the next section to use ray logs cluster? 

```
ray logs cluster raylet.out --node-ip <NODE_IP> --follow
```

**Review by rkooo567 — APPROVED** [review]
Can you update the PR description?

**Comment by rkooo567** [comment]
Can you ping the doc owner?

File: doc/source/ray-observability/state/state-api.rst [hunk 2]
```diff
@@ -482,6 +487,27 @@ E.g., Stream a log file from a node
...
+
+.. tabbed:: Python SDK
+
+    .. code-block:: python
+
+        from ray.experimental.state.api import get_log
+
+        # Node IP could be retrieved from list_nodes() or ray.nodes()
+        # You could get the actor's ID from the actor's handle, e.g.
+        # `your_actor._actor_id.hex()` or with `ray list actors`.
```
Comment (pcmoritz) [inline doc/source/ray-observability/state/state-api.rst:507]: Hmm, `_actor_id` is not a public method, do we have a public method here?

Comment (rickyyx) [reply doc/source/ray-observability/state/state-api.rst:507]: Hmm, I don't think there is. Let me maybe rephrase this as of now.

Comment (rickyyx) [reply doc/source/ray-observability/state/state-api.rst:507]: Asked in the core team and see if we want to add such util function.

Comment (pcmoritz) [reply doc/source/ray-observability/state/state-api.rst]: Yes, to make this really effective, and integrate state observability with the Ray API, I think a public method like this would be very useful.

File: doc/source/ray-observability/state/state-api.rst [hunk 3]
```diff
@@ -482,6 +487,26 @@ E.g., Stream a log file from a node
...
+
+        ray logs actor --id=<ACTOR_ID> --follow
+
+.. tabbed:: Python SDK
+
+    .. code-block:: python
+
+        from ray.experimental.state.api import get_log
+
+        # Node IP could be retrieved from list_nodes() or ray.nodes()
```
Comment (pcmoritz) [inline doc/source/ray-observability/state/state-api.rst:505]: Not sure why the reference to the Node IP is here :)

**Review by pcmoritz — APPROVED** [review]
